File: fooadgram-app/models/content/dto/RecipeResponseDto.ts
import {IngredientResponseDto} from './IngredientResponseDto';

export interface RecipeResponseDto {
    id: string;
    title: string;
    description: string;
    ingredients: IngredientResponseDto[];
    instructions: string[];
    preparationTime: number;
    cookingTime: number;
    servings: number;
    difficulty: string;
    cuisine: string;
    mealType: string;
    dietaryRestrictions?: string[];
    createdAt: string;
    updatedAt: string;
} 

File: api/clients/src/main/java/com/foodagram/clients/content/dto/RecipeResponseDto.java
package com.foodagram.clients.content.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RecipeResponseDto {
    private UUID id;
    private UUID postId;
    private String title;
    private String description;
    private List<IngredientResponseDto> ingredients;
    private List<String> instructions;
    private String cuisineType;
    private String difficultyLevel;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
} 
